core/components/commerce_timeslots/src/Admin/Schedule/Form.php

```diff
@@ -60,44 +60,45 @@ public function getFields(array $options = array())
             'label' => $this->adapter->lexicon('commerce.name'),
         ]);
 
-        // Requires the Scheduler service in Commerce 1.3+
-        $root = $this->commerce->config['core_path'];
-        $path = $root . 'src/Services/Scheduler/crontime';
-        if (file_exists($path)) {
-
-            $fields[] = new SectionField($this->commerce, [
-                'label' => $this->adapter->lexicon('commerce_timeslots.repeat_schedule'),
-                'description' => $this->adapter->lexicon('commerce_timeslots.repeat_schedule_desc'),
-            ]);
-
-            // Grab all TimeSlots shipping methods
-            $methods = $this->adapter->getCollection(\comShippingMethod::class, [
-                'class_key' => \TimeSlotsShippingMethod::class
-            ]);
-
-            if (!empty($methods)) {
-                foreach ($methods as $method) {
-                    $id = $method->get('id');
-
-                    // Generate descriptions to display any days already set by other schedules
-                    $days = $this->getUnavailableDays($id);
-
-                    $desc = '';
-                    foreach ($days as $k => $v) {
-                        $desc .= $this->adapter->lexicon('commerce_timeslots.schedule') .
-                            ' ' . $k . ': (' . implode(', ', $v) . ') ';
-                    }
 
-                    $fields[] = new SelectMultipleField($this->commerce, [
-                        'name' => 'repeat_days_' . $id,
-                        'label' => $this->adapter->lexicon('commerce_timeslots.shipping_method') . ': ' . $method->get('name'),
-                        'description' => $desc,
-                        'value' => $this->getCurrentDays($id),
-                        'options' => $this->getAvailableDays($id)
-                    ]);
+        $sectionLabel = $this->adapter->lexicon('commerce_timeslots.repeat_schedule');
+        if (!$this->commerce->isSchedulerActive()) {
```

Comment:
Hmm this will error out on 1.2 or before due to the undefined method, maybe something more like this?

```
if (!method_exists($this->commerce, 'isSchedulerActive')) { upgrade to 1.3  } 
elseif (!$this->commerce->isSchedulerActive()) { setup the scheduler }
```

Could also check for the scheduler() method itself, might be cleaner.

Reply:
Oof! not sure how I missed that 🤦🏻 

Good idea re scheduler method!